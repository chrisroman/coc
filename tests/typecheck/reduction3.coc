let idT = [a:*][b:a]a in
let id1 = (\c:*)(\d:c)d in
((id1 idT) id1)
