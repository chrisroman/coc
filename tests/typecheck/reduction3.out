let idT = [a: *] [b: a] a in 
let id1 = (λc: *) (λd: c) d in 
((id1 idT) id1)
{idT} typecheck_single_term [a: *] [b: a] a
Type: *
{id1} typecheck_single_term (λc: *) (λd: c) d
Type: [c: *] [d: c] c
typecheck_single_term (((λc: *) (λd: c) d [a: *] [b: a] a) (λc: *) (λd: c) d)
Type: [a: *] [b: a] a
Value: (λc: *) (λd: c) d
